msprime/trees.py

```diff
@@ -1779,3 +1779,60 @@ def newick_trees(self, precision=3, breakpoints=None, Ne=1):
         raise NotImplementedError(
             "This method is no longer supported. Please use the SparseTree.newick"
             " method instead")
+
+
+def compute_mutation_parent(sites, mutations, nodes=None, edges=None, ts=None):
+    """
+    (Re-)compute the `parent` column of a MutationTable.
+
+    :param SiteTable sites: The SiteTable for the tree sequence.
+    :param MutationTable mutations: The MutationTable to compute the parent
+        column for.
+    :param NodeTable nodes: The NodeTable for the tree sequence.
+    :param EdgeTable edges: The EdgeTable for the tree sequence.
+    :param TreeSequence ts: If provided, this contains the Node and Edge
+        relationships that SiteTable refers to.  If it is not provided,
+        `nodes` and `edges` will be loaded (internally) into a TreeSequence.
+    """
+    if mutations.num_rows == 0:
+        return []
+    if ts is None:
+        ts = load_tables(nodes=nodes, edges=edges)
+    new_parent = np.repeat(-1, [mutations.num_rows])
+    # sites are ordered by position,
+    #  and mutations by site.
+    mut_start = 0
+    cur_site = mutations.site[mut_start]
+    cur_pos = sites.position[cur_site]
+    num_muts = 1
+    while ((mut_start + num_muts < mutations.num_rows)
+           and (mutations.site[mut_start + num_muts] == cur_site)):
+        num_muts += 1
+    for t in ts.trees():
+        left, right = t.interval
+        if right < cur_pos:
+            next
```

Comment:
Does this work? I've never seen the next function used like this before

Reply:
hah, no. I meant `continue`.